### ISSUE TITLE
Task checkbox not updating when task is marked as incomplete

### ISSUE THREAD

**Issue by thienlnam** [Expensify/App#19086]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Break down in numbered steps
1. Create a task
2. Go back to the parent chat where the task was created
3. Click on the checkbox to mark the task as complete
4. Click on the checkbox to mark the task as incomplete

## Expected Result:
Describe what you think should've happened
The checkbox should become unchecked
## Actual Result:
Describe what actually happened
Notice that the checkbox still remains checked

## Workaround:
Clicking into the task will show the current state of the task

## Platforms:

Which of our officially supported platforms is this issue occurring on? All
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by adelekennedy** [comment]
just happened to me today - real bug, able to reproduce and couldn't find a duplicate

**Comment by Thanos30** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

We are unable to uncheck a task (make it not completed).

### What is the root cause of that problem?

The root cause of that problem is our onPress function on the Checkbox:

https://github.com/Thanos30/Expensify/blob/51d8000f626c4f02c2e20dde7de2ba1ce86e464d/src/components/ReportActionItem/TaskPreview.js#L68-L78

As you can see, if isTaskCompleted is true, we simply return the function without taking any action.

### What changes do you think we should make in order to solve the problem?

I created a `reopenTask` function, similar to our `completeTask` function, which will update the task and set the `stateNum` and `statusNum` propertied to 0 (OPEN). I won't paste the code not to make this a long one, I will post video of the fix at the end.


### What alternative solutions did you explore? (Optional)
None

Video of the fix:
https://github.com/Expensify/App/assets/32620770/5b724ed7-ed10-4da9-84a4-45fc9b803cae

**Comment by puneetlath** [comment]
Seems your right! Thanks @hoangzinh. Marking Internal.

**Comment by thienlnam** [comment]
The API command failing makes sense, but we should also add some feedback on the user end so they know that the command failed, and then disable the button until they address the error. I imagine this would look like the RBR error we se when we fail to add a comment

**Comment by jjcoffee** [comment]
@hoangzinh That's a bit of a different issue as it deals with synchronising the task across two different users.

**Comment by puneetlath** [comment]
@thienlnam re-assigning over to you as discussed [here](https://expensify.slack.com/archives/C03TQ48KC/p1685040014681889). Thanks!

**Comment by adelekennedy** [comment]
@thienlnam I saw you thumbs up the comment above - should we throw the external label on here for some contributor help?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by thienlnam** [comment]
Yeah, let me update the title - we basically need better handling for when the task cannot be marked as incomplete due to the state of the task.

Though actually this may actually also be solved in https://github.com/Expensify/App/issues/19157 so we can focus over there
